### PR TITLE
Implement basic event, person, other flashcards. Set up dummy UI to test

### LINKED ISSUE #5: As a user I can add a 'Person Flashcard'
... so that I can record the details of a person in history

# 4.3 Adding a Person Flashcard
## Format: 
add person n/PERSON_NAME bd/BIRTHDAY dd/DEATH_DAY sb/SUMMARIZED_BIOGRAPH [d/DETAIL]...

There can be 0 or multiple details added for a person.

## Example:
add person n/Severus Snape bd/9 January, 1960 dd/2 May, 1998 sb/Nicknamed Half-Blood Prince. d/His double life played an extremely important role in both of the Wizarding Wars against Voldemort. d/He met Lily and Petunia Evans when he was nine and fell deeply in love with Lily, becoming a close friend of hers.

### LINKED ISSUE #6: As a user I can add an 'Event Flashcard'
... so that I could record the details of a historical event

# 4.2 Adding an Event Flashcard
## Format: 
add event n/EVENT_NAME st/START_TIME et/END_TIME s/SUMMARY [d/DETAIL]... [p/PEOPLE]...

There can be 0 or multiple details added for an event. There can also be 0 or multiple people related to an event.

## Example:
add event n/Battle of Hogwarts st/2 May, 1998 et/2 May, 1998 s/Voldemort died d/The battle ended with a decisive victory for the Order and the D.A., with many Death Eaters and Voldemort himself dead. p/Voldemort p/Harry Potter

### LINKED ISSUE #7: As a user I can add an 'Other Flashcard'
... so that I can record the details of a historical instance that do not belong to other type of flashcards

# 4.4 Adding an Other Flashcard
## Format: 
add other t/TITLE [d/DETAIL]...

There can be 0 or multiple details added for an Other Flashcard. 

## Example:
add other t/Deathly Hallows d/The Deathly Hallows are three highly powerful magical objects supposedly created by Death and given to each of three brothers in the Peverell family. d/ Albus Dumbledore had possessed all three, but not all at once, similar to Harry, but he was never the true owner of the Cloak.

### PR DESCRIPTION
**Pull request by quinnyyy**
EventFlashcard, PersonFlashcard, OtherFlashcard inherit from abstract Flashcard class.
FlashcardFactory class can be used to get instance of flashcard.
Simple Ui class and main class.

This isn't very elegant so if you have suggestions please share!

Resolves #5  , Resolves #6 , Resolves #7

### CODE REVIEW

File: src/main/java/seedu/duke/flashcard/FlashcardFactory.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package seedu.duke.flashcard;
+
+import seedu.duke.exceptions.UnrecognizedFlashcardTypeException;
+
+public class FlashcardFactory {
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/duke/flashcard/FlashcardFactory.java]: Nice use of factory design pattern!

File: src/main/java/seedu/duke/flashcard/PersonFlashcard.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import seedu.duke.ui.Ui;
+
+import java.util.ArrayList;
+
+public class PersonFlashcard extends Flashcard {
+    private String birthDate;
+    private String deathDate;
+
+    public static PersonFlashcard createPersonFlashcard() {
+        Ui ui = new Ui();
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/duke/flashcard/PersonFlashcard.java]: Do we really need to instantiate a new UI every time we create a new flashcard? Considering having the UI object passed in to the createPersonFlashcard() method.

File: src/main/java/seedu/duke/flashcard/OtherFlashcard.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package seedu.duke.flashcard;
+
+import seedu.duke.ui.Ui;
+
+import java.util.ArrayList;
+
+public class OtherFlashcard extends Flashcard {
+    public static OtherFlashcard createOtherFlashcard() {
+        Ui ui = new Ui();
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/duke/flashcard/OtherFlashcard.java]: Consider passing in the UI as an argument

File: src/main/java/seedu/duke/flashcard/EventFlashcard.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import seedu.duke.ui.Ui;
+
+import java.util.ArrayList;
+
+public class EventFlashcard extends Flashcard {
+    private String startDate;
+    private String endDate;
+
+    public static EventFlashcard createEventFlashcard() {
+        Ui ui = new Ui();
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/duke/flashcard/EventFlashcard.java]: Consider passing in the UI as an argument

**Review by Zhilin-Huang — CHANGES_REQUESTED** [review]
Some checks have failed likely due to code style - refer to [https://github.com/AY1920S2-CS2113-T14-1/tp/pull/14/checks](https://github.com/AY1920S2-CS2113-T14-1/tp/pull/14/checks)

File: src/main/java/seedu/duke/flashcard/PersonFlashcard.java (hunk 2 above)
Comment (quinnyyy) [reply src/main/java/seedu/duke/flashcard/PersonFlashcard.java]: Good idea. I will make this change

**Review by Zhilin-Huang — COMMENTED** [review]
The code looks good to me! But I think the `runtest.sh` file needs to be updated to use `Main.java` instead of `Duke.java` so that the CI tests will not fail

**Comment by quinnyyy** [comment]
> The code looks good to me! But I think the `runtest.sh` file needs to be updated to use `Main.java` instead of `Duke.java` so that the CI tests will not fail

just changed this. looks like the checks are passing now

**Review by Zhilin-Huang — APPROVED** [review]

**Comment by Zhilin-Huang** [comment]
> > The code looks good to me! But I think the `runtest.sh` file needs to be updated to use `Main.java` instead of `Duke.java` so that the CI tests will not fail
> 
> just changed this. looks like the checks are passing now

LGTM

**Pull request merged**